libshaderc_spvc/src/spvcir_pass.cc

```diff
@@ -19,6 +19,43 @@
 namespace spvtools {
 namespace opt {
 
+void SpvcIrPass::make_constant_null(uint32_t id, uint32_t type) {
+  auto &constant_type = get<spirv_cross::SPIRType>(type);
+
+  if (constant_type.pointer) {
+    auto &constant = set<spirv_cross::SPIRConstant>(id, type);
+    constant.make_null(constant_type);
+  } else if (!constant_type.array.empty()) {
+    assert(constant_type.parent_type &&
+           "constant type parent shouldn't be empty");
+    uint32_t parent_id = ir_->increase_bound_by(1);
+    make_constant_null(parent_id, constant_type.parent_type);
+
+    assert(constant_type.array_size_literal.back() &&
+           "Array size of OpConstantNull must be a literal.");
+
+    spirv_cross::SmallVector<uint32_t> elements(constant_type.array.back());
+    for (uint32_t i = 0; i < constant_type.array.back(); i++)
+      elements[i] = parent_id;
+    set<spirv_cross::SPIRConstant>(id, type, elements.data(),
+                                   uint32_t(elements.size()), false);
+  } else if (!constant_type.member_types.empty()) {
+    uint32_t member_ids =
```

Comment:
Is this the struct case only?
Or does it cover all the non-array composites:  struct, vector, and matrix